src/quiz.js

```diff
@@ -7,16 +7,22 @@ import UserService from "./service/UserService";
 
 const quizUi = new QuizUI(document);
 const wordService = new WordService();
-const userId = localStorage.getItem(CURR_USER_KEY);
+const userId = JSON.parse(localStorage.getItem(CURR_USER_KEY));
 const userService = new UserService();
+let wordsLeftToDo;
+let totalWords;
+
+wordService.getTodayWordByUserId(userId).then((wordArray) => {
+  totalWords = wordArray.length;
+});
 
 userService.getUserById(userId).then((user) => {
   let correctWord;
   let wordCount = 0;
 
   async function* wordGenerator() {
     while (true) {
-      const wordArray = await wordService.getTodayWordByUserId(JSON.parse(localStorage.getItem(CURR_USER_KEY)));
+      const wordArray = await wordService.getTodayWordByUserId(userId);
```

Comment:
you can put `wordArray` as a global constant so you only have to get it once instead of retrieving it multiple times.

Reply:
The `wordArray` changes so you can't just get it once.